src/tests/system/tests/test_proxy.py

```diff
@@ -67,3 +67,184 @@ def test_proxy__lookup_user_using_pam_ldap_and_nslcd_with_proxy_fast_alias_enabl
 
     log = client.fs.read(client.sssd.logs.domain())
     assert "ldb_modify failed: [Invalid attribute syntax]" not in log, "'ldb_modify failed' message found in logs!"
+
+
+@pytest.mark.importance("low")
+@pytest.mark.topology(KnownTopology.LDAP)
+def test_proxy__multiple_domains_with_legacy_nss_library(client: Client, ldap: LDAP):
+    """
+    :title: Proxy Provider with Legacy NSS Library and Multiple Domains
+    :setup:
+        1. Configure SSSD with a proxy provider
+        2. Define two domains in sssd.conf
+        3. Configure distinct test users with unique UIDs
+        4. Set use_fully_qualified_names = True in sssd.conf
+    :steps:
+        1. Start SSSD
+        2. Verify FQN resolution
+        3. Test authentication
+        4. Verify system sees them as distinct users
+    :expectedresults:
+        1. SSSD starts without errors
+        2. Each FQN resolves to its correct distinct user entry
+        3. SSH authentication succeeds for both users
+        4. System maintains clear separation between domain users
+    :customerscenario: False
+    """
+    # Setup domains and users
+    ou_domain1 = ldap.ou("domain1").add()
+    user1 = ldap.user("user1", basedn=ou_domain1).add(uid=5000, gid=5000, password="Secret123")
+
+    ou_domain2 = ldap.ou("domain2").add()
+    user2 = ldap.user("user2", basedn=ou_domain2).add(uid=5001, gid=5001, password="Secret123")
+
+    # Configure SSSD with two proxy domains
+    client.sssd.common.proxy("ldap", ["id", "auth"], server_hostname=ldap.host.hostname, domain="domain1.example.com")
+    client.sssd.dom("domain1.example.com")["use_fully_qualified_names"] = "True"
+
+    client.sssd.common.proxy("ldap", ["id", "auth"], server_hostname=ldap.host.hostname, domain="domain2.example.com")
+    client.sssd.dom("domain2.example.com")["use_fully_qualified_names"] = "True"
+
+    # Set domain order with domain1 first
+    client.sssd.sssd.update(domains="domain1.example.com, domain2.example.com")
+
+    # Restart services
+    client.sssd.svc.restart("nslcd")
+    client.sssd.restart()
+
+    # Test FQN resolution
+    result1 = client.tools.getent.passwd(f"{user1.name}@domain1.example.com")
+    assert result1 is not None, "User1 not found with FQN"
+    assert result1.uid == 5000, "User1 has wrong UID"
+    assert result1.home == "/home/user1", "User1 has wrong home directory"
+
+    result2 = client.tools.getent.passwd(f"{user2.name}@domain2.example.com")
+    assert result2 is not None, "User2 not found with FQN"
+    assert result2.uid == 5001, "User2 has wrong UID"
+    assert result2.home == "/home/user2", "User2 has wrong home directory"
+
+    # Test authentication
+    assert client.auth.ssh.password(f"{user1.name}@domain1.example.com", password="Secret123"), "User1 login failed"
+
+    assert client.auth.ssh.password(f"{user2.name}@domain2.example.com", password="Secret123"), "User2 login failed"
```

Comment:
While I agree that the extra line between these two makes it a little easier to read, I think it would make the code more consistent to remove it.   Since you don't have blank lines between other lines in a "section" separated by line comments, I think it makes sense to remove the blank line here too.  

If you disagree and in some circumstances makes it harder to read, feel free to ignore this comment.

Reply:
Agreed and changed